joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -58,71 +159,113 @@ class AstCreator(filename: String, global: Global)
   protected def lineEnd(node: TerminalNode): Option[Integer]   = None
   protected def columnEnd(node: TerminalNode): Option[Integer] = None
 
-  def astForVariableIdentifierContext(ctx: VariableIdentifierContext, varType: String): Ast = {
+  private def registerType(typ: String): String = {
+    if (typ != Defines.Any) {
+      global.usedTypes.putIfAbsent(typ, true)
+    }
+    typ
+  }
+  def astForVariableIdentifierContext(ctx: VariableIdentifierContext): Ast = {
     val terminalNode = ctx.children.asScala.map(_.asInstanceOf[TerminalNode]).head
     val token        = terminalNode.getSymbol
     val variableName = token.getText
-    val node         = identifierNode(terminalNode, variableName, variableName, varType, List(varType))
+    val node         = createIdentiferWithScope(terminalNode, variableName, variableName, Defines.Any, List[String]())
+    setIdentiferInScope(node)
     Ast(node)
   }
 
-  def astForSingleLeftHandSideContext(ctx: SingleLeftHandSideContext, rhsRetType: String): Ast = ctx match {
+  def astForSingleLeftHandSideContext(ctx: SingleLeftHandSideContext): Ast = ctx match {
     case ctx: VariableIdentifierOnlySingleLeftHandSideContext =>
-      astForVariableIdentifierContext(ctx.variableIdentifier(), rhsRetType)
-    case ctx: PrimaryInsideBracketsSingleLeftHandSideContext => astForPrimaryContext(ctx.primary())
+      astForVariableIdentifierContext(ctx.variableIdentifier())
+    case ctx: PrimaryInsideBracketsSingleLeftHandSideContext =>
+      val primaryAst = astForPrimaryContext(ctx.primary())
+      val argsAst    = astForArgumentsContext(ctx.arguments())
+      val callNode = NewCall()
+        .name(Operators.indexAccess)
+        .code(Operators.indexAccess)
```

Comment:
The `code` field here is incorrect, it should contain the code of the entire expression